src/main/java/de/tum/in/www1/artemis/service/ModelingExerciseService.java

```diff
@@ -29,4 +40,30 @@ public ModelingExercise findOne(Long exerciseId) {
         log.debug("Request to get Modeling Exercise : {}", exerciseId);
         return modelingExerciseRepository.findById(exerciseId).orElseThrow(() -> new EntityNotFoundException("Exercise with id: \"" + exerciseId + "\" does not exist"));
     }
+
+    /**
+     * Search for all modeling exercises fitting a {@link PageableSearchDTO search query}. The result is paged,
+     * meaning that there is only a predefined portion of the result returned to the user, so that the server doesn't
+     * have to send hundreds/thousands of exercises if there are that many in Artemis.
+     *
+     * @param search The search query defining the search term and the size of the returned page
+     * @param user The user for whom to fetch all available exercises
+     * @return A wrapper object containing a list of all found exercises and the total number of pages
+     */
+    public SearchResultPageDTO<ModelingExercise> getAllOnPageWithSize(final PageableSearchDTO<String> search, final User user) {
+        var sorting = Sort.by(TextExercise.TextExerciseSearchColumn.valueOf(search.getSortedColumn()).getMappedColumnName());
```

Comment:
Yeah, I wanted to move this function but it slipped my mind. Thank you :D I moved it to Exercise.java and renamed it since it is used by all exercises which allow imports.
See: 8982dccea8a48d445f78ff1f8df08d4391bc9416